### PR TITLE
save SQL group when writing extra heuristics statistics 

### PR DESCRIPTION
**Pull request by jgaleotti**

### CODE REVIEW

File: core/src/main/kotlin/org/evomaster/core/database/ReplaceValuesDeParser.kt [hunk 1]
```diff
@@ -0,0 +1,46 @@
+package org.evomaster.core.database
+
+import net.sf.jsqlparser.expression.*
+import net.sf.jsqlparser.util.deparser.ExpressionDeParser
+
+
+class ReplaceValuesDeParser() :
```
Comment (arcuri82) [inline core/src/main/kotlin/org/evomaster/core/database/ReplaceValuesDeParser.kt:7]: add some comment about what this class is for

File: core/src/main/kotlin/org/evomaster/core/search/service/ExtraHeuristicsLogger.kt [hunk 2]
```diff
@@ -21,45 +28,61 @@ class ExtraHeuristicsLogger {
...
+        Files.createFile(filePath)
+    }
 
+    private fun getExtraHeuristicsFilePath() = Paths.get(config.extraHeuristicsFile).toAbsolutePath()
 
     fun writeHeuristics(list: List<HeuristicEntryDto>, actionId: Int) {
-        if (!config.writeExtraHeuristicsFile) {
-            return
-        }
+        if (config.writeExtraHeuristicsFile) {
```
Comment (arcuri82) [inline core/src/main/kotlin/org/evomaster/core/search/service/ExtraHeuristicsLogger.kt:50]: revert. usually it is better to return immediately on a negated condition, as it avoid a long nested block

File: core/src/main/kotlin/org/evomaster/core/search/service/ExtraHeuristicsLogger.kt [hunk 3]
```diff
@@ -21,45 +28,61 @@ class ExtraHeuristicsLogger {
...
 
-        val counter = time.evaluatedIndividuals
+            val counter = time.evaluatedIndividuals
+            val lines = list.map {
+                val group = if (it.type == HeuristicEntryDto.Type.SQL) {
+                    try {
+                        "\"${removeAllConstants(it.id)}\""
```
Comment (arcuri82) [inline core/src/main/kotlin/org/evomaster/core/search/service/ExtraHeuristicsLogger.kt:56]: as this is going to be put in a CSV file, we need to make sure that we properly handle any `,` character

**Review by arcuri82 — CHANGES_REQUESTED** [review]

File: core/src/main/kotlin/org/evomaster/core/database/ReplaceValuesDeParser.kt (hunk 1 above)
Comment (jgaleotti) [reply core/src/main/kotlin/org/evomaster/core/database/ReplaceValuesDeParser.kt:7]: ok

File: core/src/main/kotlin/org/evomaster/core/search/service/ExtraHeuristicsLogger.kt (hunk 2 above)
Comment (jgaleotti) [reply core/src/main/kotlin/org/evomaster/core/search/service/ExtraHeuristicsLogger.kt:50]: ok.

File: core/src/main/kotlin/org/evomaster/core/search/service/ExtraHeuristicsLogger.kt (hunk 3 above)
Comment (jgaleotti) [reply core/src/main/kotlin/org/evomaster/core/search/service/ExtraHeuristicsLogger.kt:56]: One question, what about "org.evomaster.client.java.controller.api.dto.HeuristicEntryDto#id" 
How the "," character is handled here? 
It looks as the same problem concerns both it.id and  removeAllConstants(it.id), isn't it?

Comment (arcuri82) [reply core/src/main/kotlin/org/evomaster/core/search/service/ExtraHeuristicsLogger.kt:56]: hi. yes, technically it is the same issue. maybe we haven't encountered it before. anyway, if issue happens, we ll see problem as soon as we try to load the generated CSV file, due to column mismatches.
should check how `,` commas are escaped in CSV

**Pull request merged**